src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -79,6 +79,25 @@ void _buildSHDRecordHelperVisitor(std::optional<T> value, String&, shd::SKVRecor
     }
 }
 
+// Convert skv element vector to k2 element vector by convert function
+template <class K2Type, class SHType, typename Fn>
+auto shVectorToK2(std::vector<SHType>&  shElems, Fn convertFn) {
```

Comment:
These two helper functions end up consuming the elements in the vector. You should instead require the caller to call you with rvalue ref (i.e. shVectorToK2(std::move(myvec))

Reply:
Done.